### PR TITLE
TINY-9497: Enabled icon colors

### PR DESCRIPTION
**Pull request by HAFRMO**
Related Ticket: 

Description of Changes:
Icons did not have the ability to get colors, so added that. Same mechanism as toolbar buttons. Also exposed the getUiRoot function in test framework, as it can often be used.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/bridge/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 - Added optional `select`-property to `colorswatch`-type `fancymenuitem` to allow setting colors as selected. #TINY-9395
+- Added new `setIconColor`-function to `nestedmenuitemspec`. #TINY-9497
```
Comment (spocke) [inline modules/bridge/CHANGELOG.md:9]:
```suggestion
- Added new `setIconFill`-function to `nestedmenuitemspec`. #TINY-9497
```

File: modules/tinymce/src/themes/silver/main/ts/ui/menus/item/build/NestedMenuItem.ts [hunk 2]
```diff
@@ -14,7 +15,12 @@ const renderNestedItem = (spec: Menu.NestedMenuItem, itemResponse: ItemResponse,
   const caret = downwardsCaret ? renderDownwardsCaret(providersBackstage.icons) : renderSubmenuCaret(providersBackstage.icons);
   const getApi = (component: AlloyComponent): Menu.NestedMenuItemInstanceApi => ({
     isEnabled: () => !Disabling.isDisabled(component),
-    setEnabled: (state: boolean) => Disabling.set(component, !state)
+    setEnabled: (state: boolean) => Disabling.set(component, !state),
+    setIconFill: (id, value) => {
+      SelectorFind.descendant(component.element, 'svg path[id="' + id + '"], rect[id="' + id + '"]').each((underlinePath) => {
```
Comment (spocke) [inline modules/tinymce/src/themes/silver/main/ts/ui/menus/item/build/NestedMenuItem.ts:20]:
```suggestion
      SelectorFind.descendant(component.element, `svg path[id="${id}"], rect[id="${id}"]`).each((underlinePath) => {
```

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The `onSetup` api function would not run when defining custom group toolbar button. #TINY-9496
 - An element could be dropped onto the decendants of a noneditable element. #TINY-9364
 - Checkmark did not show in menu colorswatches. #TINY-9395
+- The foreground and background menu icons would not properly update to display the last used color. #TINY-9497
```
Comment (spocke) [inline modules/tinymce/CHANGELOG.md:27]: You also introduced a new `setIconFill` api to all nested menu items so a separate line for that would be nice.

**Review by spocke — APPROVED** [review]

File: modules/bridge/CHANGELOG.md (hunk 1 above)
Comment (HAFRMO) [reply modules/bridge/CHANGELOG.md:9]: Woups.

File: modules/tinymce/CHANGELOG.md [hunk 4]
```diff
@@ -24,6 +24,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Added new `setIconFill` function to `nestedmenuitemspec` #TINY-9497
```
Comment (spocke) [inline modules/tinymce/CHANGELOG.md:28]:
```suggestion
- Added new `setIconFill` function to `NestedMenuItemInstanceApi`. #TINY-9497
```
or
```suggestion
- Added new `setIconFill` API to nested menu items. #TINY-9497
```

**Review by spocke — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 4 above)
Comment (HAFRMO) [reply modules/tinymce/CHANGELOG.md:28]: Fair enough.

File: modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideColorSwatchMenuTest.ts [hunk 5]
```diff
@@ -300,6 +306,19 @@ describe('browser.tinymce.themes.silver.skin.OxideColorSwatchMenuTest', () => {
...
+  it('TINY-9497: Opening the menu with different colors should display in the menu', async () => {
+    const editor = hook.editor();
+
+    editor.setContent('<p>black</p>');
+    TinySelections.setSelection(editor, [ 0, 0 ], 1, [ 0, 0 ], 2, true);
+    await pOpenAndGetMenuColorMenu(editor);
+    UiFinder.pWaitFor('The color should be black in the icon', TinyUiActions.getUiRoot(editor), 'path[id="tox-icon-highlight-bg-color__color"][fill="#000000"]');
+    TinyUiActions.clickOnUi(editor, '[role="menuitemradio"][title="red"]');
+    await pOpenAndGetMenuColorMenu(editor);
+    UiFinder.pWaitFor('The color should be black in the icon', TinyUiActions.getUiRoot(editor), 'path[id="tox-icon-highlight-bg-color__color"][fill="#E03E2D"]');
```
Comment (vpyshnenko) [inline modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideColorSwatchMenuTest.ts:321]:
```suggestion
    UiFinder.pWaitFor('The color should be red in the icon', TinyUiActions.getUiRoot(editor), 'path[id="tox-icon-highlight-bg-color__color"][fill="#E03E2D"]');
```

**Review by vpyshnenko — APPROVED** [review]

**Review by TheSpyder — CHANGES_REQUESTED** [review]
See comments in JIRA

**Comment by HAFRMO** [comment]
JIRA issue solved.

**Review by TheSpyder — APPROVED** [review]

**Pull request merged**